tests/migrations/test_autogen_migration.py

```diff
@@ -0,0 +1,160 @@
+    cols = [
+        "{ name: project_id, type: UInt, args: { size: 64 } }",
+        "{ name: timestamp, type: DateTime }",
+        "{ name: event_id, type: UUID }",
+    ]
+    new_cols = [
+        "{ name: project_id, type: UInt, args: { size: 64 } }",
+        "{ name: timestamp, type: DateTime }",
+        "{ name: newcol1, type: DateTime }",
+        "{ name: event_id, type: UUID }",
+        "{ name: newcol2, type: UInt, args: { schema_modifiers: [nullable], size: 8 } }",
+    ]
+    forwardops, backwardsops = generate_migration_ops(
+        mockstoragewithcolumns(cols), mockstoragewithcolumns(new_cols)
+    )
+    expected_forward = [
+        AddColumn(
+            storage_set=StorageSetKey("events"),
+            table_name="errors_local",
+            column=Column("newcol1", DateTime()),
+            after="timestamp",
+            target=OperationTarget.LOCAL,
+        ),
+        AddColumn(
+            storage_set=StorageSetKey("events"),
+            table_name="errors_dist",
+            column=Column("newcol1", DateTime()),
+            after="timestamp",
+            target=OperationTarget.DISTRIBUTED,
+        ),
+        AddColumn(
+            storage_set=StorageSetKey("events"),
+            table_name="errors_local",
+            column=Column(
+                "newcol2", UInt(size=8, modifiers=MigrationModifiers(nullable=True))
+            ),
+            after="event_id",
+            target=OperationTarget.LOCAL,
+        ),
+        AddColumn(
+            storage_set=StorageSetKey("events"),
+            table_name="errors_dist",
+            column=Column(
+                "newcol2", UInt(size=8, modifiers=MigrationModifiers(nullable=True))
+            ),
+            after="event_id",
+            target=OperationTarget.DISTRIBUTED,
+        ),
+    ]
+    expected_backwards = [
+        DropColumn(
+            storage_set=StorageSetKey("events"),
+            table_name="errors_dist",
+            column_name="newcol2",
+            target=OperationTarget.DISTRIBUTED,
+        ),
+        DropColumn(
+            storage_set=StorageSetKey("events"),
+            table_name="errors_local",
+            column_name="newcol2",
+            target=OperationTarget.LOCAL,
+        ),
+        DropColumn(
+            storage_set=StorageSetKey("events"),
+            table_name="errors_dist",
+            column_name="newcol1",
+            target=OperationTarget.DISTRIBUTED,
+        ),
+        DropColumn(
+            storage_set=StorageSetKey("events"),
+            table_name="errors_local",
+            column_name="newcol1",
+            target=OperationTarget.LOCAL,
+        ),
+    ]
+    assert forwardops == expected_forward and backwardsops == expected_backwards
+
+
+def test_modify_column() -> None:
+    cols = [
+        "{ name: timestamp, type: DateTime }",
+    ]
+    new_cols = [
+        "{ name: timestamp, type: UUID }",
+    ]
+    with pytest.raises(ValueError):
```

Comment:
would be good to check that this fails for the reason you expect and that the error message is correct and instructive to the user